src/client/core/utils/array.js

```diff
@@ -62,6 +62,10 @@ export function reduce (arr, callback, initialValue) {
     return nativeReduce.call(arr, callback, initialValue);
 }
 
+export function unshift (arr, ...items) {
```

Comment:
It's a nitpick, but maybe it's better to use the native `Array.prototype.unshift` here?

Moreover, I feel that we can generalize and simplify this code:
```
const ARRAY_PROTO = Array.prototype;

function createWrapper (method) {
    return (...args) => ARRAY_PROTO[method.name].call(...args);
} 

export const unshift = createWrapper(ARRAY_PROTO.unshift);
//...

```

Of course, we can use just function names instead of function objects, but I feel that we should convert this file to TypeScript and do something to be sure that we use correct property names, e.g. `function createWrapper (methodName: keyof Array): Function {`

Also it's interesting why do we use a custom `find` function. I cannot remember the reason why we did it instead of just using a wrapper like for other functions.

Reply:
Also, it's better to take the native Array methods from the `hammerhead.nativeMethods`.

Reply:
Seems like we create our own `find` method for IE. There still isn't this one.

Reply:
Thank you both for your comments I will rethink this file